simplesat/tests/test_rules_generator.py

```diff
@@ -0,0 +1,48 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import io
+import unittest
+
+from enstaller.new_solver import Pool
+
+from ..rules_generator import RuleType, RulesGenerator
+from ..test_utils import Scenario
+
+
+class TestRulesGenerator(unittest.TestCase):
+
+    def test_prefer_installed(self):
+
```

Comment:
We generally follow the convention of separating a test into `Given`, `When` and `Then` sections (delimited by comments, see other tests).